consensus/clique/src/main/java/org/hyperledger/besu/consensus/clique/CliqueHelpers.java

```diff
@@ -90,4 +92,36 @@ private static int minimumBlocksSincePreviousSigning(
     final int signerLimit = (validatorCount / 2) + 1;
     return signerLimit - 1;
   }
+
+  public static void installCliqueBlockChoiceRule(
+      final Blockchain blockchain, final CliqueContext cliqueContext) {
+    blockchain.setBlockChoiceRule(
+        blockchain
+            // most total difficulty
+            .getBlockChoiceRule()
+            // shortest chain
+            .thenComparing(Comparator.comparing(ProcessableBlockHeader::getNumber).reversed())
+            // proposer with least recent in-turn block
```

Comment:
should this be "most recent" or is the PR description wrong?

Reply:
this flopped around several times.  current eip says maximum of the formula, so it's an inverted comparator.